test/listen.spec.js

```diff
@@ -0,0 +1,64 @@
+/* eslint-env mocha */
+
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const Stardust = require('../src')
+
+const { createPeer, SERVER_URL } = require('./utils')
+const mockUpgrader = {
+  upgradeInbound: maConn => maConn,
+  upgradeOutbound: maConn => maConn
+}
+
+describe('listen', () => {
+  let stardust
+
+  beforeEach(async () => {
+    const [libp2p] = await createPeer()
+    stardust = new Stardust({ upgrader: mockUpgrader, libp2p })
+  })
+
+  it('listen, check for promise', async () => {
```

Comment:
improve description about what's being tested
```suggestion
  it('should be able to listen on a valid server', async () => {
```